gtsam/linear/NoiseModel.cpp

```diff
@@ -135,14 +135,25 @@ bool Gaussian::equals(const Base& expected, double tol) const {
   const Gaussian* p = dynamic_cast<const Gaussian*> (&expected);
   if (p == nullptr) return false;
   if (typeid(*this) != typeid(*p)) return false;
-  //if (!sqrt_information_) return true; // ALEX todo;
+  //TODO(Alex);
+  //if (!sqrt_information_) return true;
   return equal_with_abs_tol(R(), p->R(), sqrt(tol));
 }
 
+/* ************************************************************************* */
+Matrix Gaussian::covariance() const {
+  // Uses a fast version of `covariance = information().inverse();`
+  Matrix R = this->R();
+  Matrix I = Matrix::Identity(R.rows(), R.cols());
+  // Fast inverse of upper-triangular matrix R using forward-substitution
+  Matrix Rinv = R.triangularView<Eigen::Upper>().solve(I);
```

Comment:
Actually, https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html says we should use LLT. as does https://stackoverflow.com/questions/38640563/eigen-efficient-inverse-of-symmetric-positive-definite-matrix

Reply:
Umm I don't understand. `R` is already upper triangular, so why decompose again? Since this is my call, I am happy sticking to the current implementation.